### DISCUSSION TITLE
[Question]: Are OpenVino and TF Lite models different?

### DISCUSSION THREAD

**Discussion by jamesonuk** [blakeblackshear/frigate#12636] [Ask A Question]
### What is your question:

This is something that came up elsewhere that someone was saying when they switched from OpenVino on their GPU to the TF Lite model on a Coral that the object detection accuracy rating was consistently lower using the Coral than the GPU.

I had (perhaps naively) assumed that there was a single model that had been converted into different formats.

Are the models actually different and have been trained on different things?

If they are the same model is there any reason as to why this user would see different results using one over the other?

**Comment by NickM-27** [comment]
They are different, trains on the same COCO dataset but with different architectures. 

The Coral model is MobileDet

https://github.com/blakeblackshear/frigate/blob/7c39b176ac486483e1e45188e15745fa1eaee467/docker/main/Dockerfile#L72

while the openvino is SSDMobileNet V2

https://github.com/blakeblackshear/frigate/blob/7c39b176ac486483e1e45188e15745fa1eaee467/docker/main/Dockerfile#L76

**Reply by jamesonuk** [reply]
> They are different, trains on the same COCO dataset but with different architectures.

I see that the OpenVino one is coming from tensorflow.org and the Coral one is coming from the google-coral github.
https://github.com/blakeblackshear/frigate/blob/7c39b176ac486483e1e45188e15745fa1eaee467/docker/main/Dockerfile#L65

but I am struggling to find much info about the models or the data they were trained on

If they are both trained on the same data is it likely they would give quite differing results?

Also I noticed there is a `project-grouper` version of the model for the coral which appears to be a bit newer than the `project-frogfish` version that is built into Frigate.   I assume there is a good reason for that.

I have played around with TensorFlow for a few work projects so would be quite interested to start digging here where I have any actual use for the models :)

**Reply by NickM-27** [reply]
> If they are both trained on the same data is it likely they would give quite differing results?

yes of course, model architecture, training weights, training methodology, etc. all have impacts on the model detection. At the very basic level you can see that the openvino model is trained on 300x300 images while the coral model is trained on 320x320 images
